README.pod
=pod

=encoding utf-8

=head1 NAME

Dist::Zilla::PluginBundle::DAGOLDEN - Dist::Zilla configuration the way DAGOLDEN does it

=head1 VERSION

version 0.049

=head1 SYNOPSIS

   # in dist.ini
   [@DAGOLDEN]

=head1 DESCRIPTION

This is a L<Dist::Zilla> PluginBundle.  It is roughly equivalent to the
following dist.ini:

   ; version provider
   [Git::NextVersion]  ; get version from last release tag
   version_regexp = ^release-(.+)$
 
   ; collect contributors list
   [ContributorsFromGit]
 
   ; choose files to include
   [Git::GatherDir]         ; everything from git ls-files
   exclude_filename = README.pod   ; skip this generated file
   exclude_filename = META.json    ; skip this generated file
 
   [PruneCruft]        ; default stuff to skip
   [ManifestSkip]      ; if -f MANIFEST.SKIP, skip those, too
 
   ; file modifications
   [OurPkgVersion]     ; add $VERSION = ... to all files
   [InsertCopyright    ; add copyright at "# COPYRIGHT"
   [PodWeaver]         ; generate Pod
   config_plugin = @DAGOLDEN ; my own plugin allows Pod::WikiDoc
 
   ; generated files
   [License]           ; boilerplate license
   [ReadmeAnyFromPod]     ; from Pod (runs after PodWeaver)
   [ReadmeAnyFromPod / ReadmeInRoo ]  ; create README.pod in repo directory
   type = pod
   filename = README.pod
   location = root
 
   ; t tests
   [Test::Compile]     ; make sure .pm files all compile
   fake_home = 1       ; fakes $ENV{HOME} just in case
 
   [Test::MinimumPerl]
   max_target_perl = 5.010
 
   [Test::ReportPrereqs] ; show prereqs in automated test output
 
   ; xt tests
   [Test::PodSpelling] ; xt/author/pod-spell.t
   [Test::Perl::Critic]; xt/author/critic.t
   [MetaTests]         ; xt/release/meta-yaml.t
   [PodSyntaxTests]    ; xt/release/pod-syntax.t
   [PodCoverageTests]  ; xt/release/pod-coverage.t
   [Test::Portability] ; xt/release/portability.t (of file name)
   options = test_one_dot = 0
   [Test::Version]     ; xt/release/test-version.t
 
   ; metadata
   [AutoPrereqs]       ; find prereqs from code
   skip = ^t::lib
 
   [Authority]
   authority = cpan:DAGOLDEN
   do_munging = 0
 
   [MinimumPerl]       ; determine minimum perl version
 
   [MetaNoIndex]       ; sets 'no_index' in META
   directory = t
   directory = xt
   directory = examples
   directory = corpus
   package = DB        ; just in case
 
   [GithubMeta]        ; set META resources
   user = dagolden
   remote = origin
   remote = github
   issues = 1
 
   [MetaProvides::Package] ; add 'provides' to META files
   meta_noindex = 1        ; respect prior no_index directives
 
   [MetaYAML]          ; generate META.yml (v1.4)
   [MetaJSON]          ; generate META.json (v2)
 
   ; build system
   [ExecDir]           ; include 'bin/*' as executables
   [ShareDir]          ; include 'share/' for File::ShareDir
   [MakeMaker]         ; create Makefile.PL
 
   ; manifest (after all generated files)
   [Manifest]          ; create MANIFEST
 
   ; copy META.json back to repo dis
   [CopyFilesFromBuild]
   copy = META.json
 
   ; before release
   [Git::Check]        ; ensure all files checked in
   allow_dirty = dist.ini
   allow_dirty = Changes
   allow_dirty = README.pod
   allow_dirty = META.json
 
   [CheckMetaResources]     ; ensure META has 'resources' data
   [CheckPrereqsIndexed]    ; ensure prereqs are on CPAN
   [CheckChangesHasContent] ; ensure Changes has been updated
   [CheckExtraTests]   ; ensure xt/ tests pass
   [TestRelease]       ; ensure t/ tests pass
   [ConfirmRelease]    ; prompt before uploading
 
   ; releaser
   [UploadToCPAN]      ; uploads to CPAN
 
   ; after release
   [Git::Commit / Commit_Dirty_Files] ; commit Changes (as released)
 
   [Git::Tag]          ; tag repo with custom tag
   tag_format = release-%v
 
   ; NextRelease acts *during* pre-release to write $VERSION and
   ; timestamp to Changes and  *after* release to add a new {{$NEXT}}
   ; section, so to act at the right time after release, it must actually
   ; come after Commit_Dirty_Files but before Commit_Changes in the
   ; dist.ini.  It will still act during pre-release as usual
 
   [NextRelease]
 
   [Git::Commit / Commit_Changes] ; commit Changes (for new dev)
 
   [Git::Push]         ; push repo to remote
   push_to = origin

=for stopwords autoprereq dagolden fakerelease pluginbundle podweaver
taskweaver uploadtocpan dist ini

=for Pod::Coverage configure mvp_multivalue_args

=head1 USAGE

To use this PluginBundle, just add it to your dist.ini.  You can provide
the following options:

=over

=item *

C<<< is_task >>> -- this indicates whether TaskWeaver or PodWeaver should be used.
Default is 0.

=item *

C<<< authority >>> -- specifies the x_authority field for pause.  Defaults to 'cpan:DAGOLDEN'.

=item *

C<<< auto_prereq >>> -- this indicates whether AutoPrereq should be used or not.  Default is 1.

=item *

C<<< fake_release >>> -- swaps FakeRelease for UploadToCPAN. Mostly useful for testing a dist.ini without risking a real release.

=item *

C<<< git_remote >>> -- where to push after release

=item *

C<<< github_issues >>> -- whether to use github issue tracker. Defaults is 1.

=item *

C<<< stopwords >>> -- add stopword for Test::PodSpelling (can be repeated)

=item *

C<<< tag_format >>> -- given to C<<< Git::Tag >>>.  Default is 'release-%v' to be more
robust than just the version number when parsing versions for
C<<< Git::NextVersion >>>

=item *

C<<< weaver_config >>> -- specifies a Pod::Weaver bundle.  Defaults to @DAGOLDEN.

=item *

C<<< version_regexp >>> -- given to C<<< Git::NextVersion >>>.  Default
is '^release-(.+)$'

=item *

C<<< no_git >>> -- bypass all git-dependent plugins

=item *

C<<< no_critic >>> -- omit Test::Perl::Critic tests

=item *

C<<< no_spellcheck >>> -- omit Test::PodSpelling tests

=item *

C<<< no_coverage >>> -- omit PodCoverage tests

=item *

C<<< no_minimum_perl >>> -- omit Test::MinimumPerl tests

=item *

C<<< no_bugtracker >>> -- DEPRECATED

=back

When running without git, CE<lt>GatherDirE<gt> is used instead of CE<lt>Git::GatherDirE<gt>,
CE<lt>AutoVersionE<gt> is used instead of CE<lt>Git::NextVersionE<gt>, and all git check and
commit operations are disabled.

This PluginBundle now supports ConfigSlicer, so you can pass in options to the
plugins used like this:

   [@DAGOLDEN]
   ExecDir.dir = scripts ; overrides ExecDir

This PluginBundle also supports PluginRemover, so dropping a plugin is as easy as this:

   [@DAGOLDEN]
   -remove = PluginIDontWant

=head1 COMMON PATTERNS

=head2 use github instead of RT

   [@DAGOLDEN]
   :version = 0.32
   AutoMetaResources.bugtracker.github = user:dagolden
   AutoMetaResources.bugtracker.rt = 0

=head1 SEE ALSO

=over

=item *

L<Dist::Zilla>

=item *

L<Dist::Zilla::Plugin::PodWeaver>

=item *

L<Dist::Zilla::Plugin::TaskWeaver>

=back

=for :stopwords cpan testmatrix url annocpan anno bugtracker rt cpants kwalitee diff irc mailto metadata placeholders metacpan

=head1 SUPPORT

=head2 Bugs / Feature Requests

Please report any bugs or feature requests through the issue tracker
at L<https://github.com/dagolden/dist-zilla-pluginbundle-dagolden/issues>.
You will be notified automatically of any progress on your issue.

=head2 Source Code

This is open source software.  The code repository is available for
public review and contribution under the terms of the license.

L<https://github.com/dagolden/dist-zilla-pluginbundle-dagolden>

  git clone https://github.com/dagolden/dist-zilla-pluginbundle-dagolden.git

=head1 AUTHOR

David Golden <dagolden@cpan.org>

=head1 CONTRIBUTORS

=over 4

=item *

Christian Walde <walde.christian@googlemail.com>

=item *

Eric Johnson <eric.git@iijo.org>

=item *

Karen Etheridge <ether@cpan.org>

=item *

Philippe Bruhat (BooK) <book@cpan.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2013 by David Golden.

This is free software, licensed under:

  The Apache License, Version 2.0, January 2004



Changes
Revision history for Dist-Zilla-PluginBundle-DAGOLDEN

{{$NEXT}}

    [CHANGED]

    - Moved Git::Contributors later in the plugin bundle so it
      can detect a minimum Perl prereqs and stop warning about
      Perl <= 5.8.6 and non-ascii characters.

0.078     2016-08-09 15:14:28-04:00 America/New_York

    [ADDED]

    - Added 'no_copy_files' option to skip Makefile.PL or cpanfile copying

0.077     2016-06-20 15:32:53-04:00 America/New_York

    [FIXED]

    - Fixed failing test under Dist::Zilla 6.005 or later

0.076     2015-07-16 11:27:10-04:00 America/New_York

    [CHANGED]

    - Swapped in SurgicalPodWeaver for PodWeaver

0.075     2015-06-12 11:40:31-04:00 America/New_York

    [CHANGED]

    - Swapped in CopyFilesFromBuild::Filtered for the unfiltered version.
      This gets rid of "generated by" lines in Makefile.PL that cause
      unnecessary file churn as Dist::Zilla is upgraded.

0.074     2015-01-22 17:07:41-05:00 America/New_York

    [CHANGED]

    - Allow dirty Makefile.PL/cpanfile and commit it before tagging.

0.073     2014-12-10 16:36:34-05:00 America/New_York

    [CHANGED]

    - Swapped out ReadmeFromPod for Pod2Readme, which is simpler and
      lighter.

0.072     2014-09-26 16:23:55-04:00 America/New_York

    [FIXED]

    - Fixed bug where version bumps weren't being committed after release

0.071     2014-09-26 11:56:23-04:00 America/New_York

    [*** INCOMPATIBLE CHANGES ***]

    - No longer copies cpanfile after build (except under auto_version);
      instead copies Makefile.PL.  With a Makefile.PL and embedded
      $VERSION, this makes a repo checkout ready for end-users without
      needing to know anything about Dist::Zilla.

0.070     2014-09-25 17:36:54-04:00 America/New_York

    [*** INCOMPATIBLE CHANGES ***]

    - Switched version handling from Git::NextVersion + OurPkgVersion to
      RewriteVerson + BumpVersionAfterRelease so that files in the repo
      will have a $VERSION.  This will require adding a $VERSION statement
      back to all modules before releasing.  Also added an "auto_version"
      option to allow AutoVersion + PkgVersion for distributions that need
      it.

    [CHANGED]

    - No longer allows dirty files at release time

    - Switched from ContributorsFromGit to Git::Contributors and updated
      Pod::Weaver::Section::Contributors prereq to match

0.069     2014-07-29 14:17:19-04:00 America/New_York

    [CHANGED]

    - Bumped Test::ReportPrereqs dependency to 0.016 because I'm an idiot

0.068     2014-07-29 13:39:29-04:00 America/New_York

    [CHANGED]

    - Bumped Test::ReportPrereqs dependency to 0.015

0.067     2014-04-17 16:43:11-04:00 America/New_York

    [ADDED]

    - Added Git::CheckFor::CorrectBranch plugin

0.066     2014-03-28 17:05:19-04:00 America/New_York

    [CHANGED]

    - Switched back to MinimumPerl; MinimumPerlFast isn't quite stable
      enough yet for me to depend on it

0.065     2014-03-23 07:48:28-04:00 America/New_York

    [CHANGED]

    - set default_jobs = 9 for RunExtraTests plugin

0.064     2014-03-16 22:28:53+01:00 Europe/Paris

    [CHANGED]

    - Switched from MinimumPerl to MinimumPerlFast

0.063     2014-03-16 17:03:18+01:00 Europe/Paris

    [CHANGED]

    - Removed ununsed autodie and Moose::AutoBox

    - set default_jobs = 9 for MakeMaker plugin

0.062     2014-02-02 21:13:10-05:00 America/New_York

    [FIXED]

    - Fixed breakage from Config::MVP 2.200007

0.061     2014-01-15 19:39:55-05:00 America/New_York

    [CHANGED]

    - Actually, we need Pod::Elemental::PerlMunger 2.000001

0.060     2014-01-15 09:35:50-05:00 America/New_York

    [CHANGED]

    - Pod weaving no longer modifies code line numbers; Pod mixed with
      code is commented rather than replaced, but Pod after code is
      removed and replaced with generated Pod

0.059     2013-11-09 21:57:32 America/New_York

    [ADDED]

    - Adds PromptIfStale to check build tool versions before release

    [CHANGED]

    - Requires Pod::Elemental::Transformer::List 0.102000
      (bullet lists that wrap with leading whitespace)

    [REMOVED]

    - Removes unused Pod::Weaver::Plugin::Encoding (since we use
      the built-in SingleEncoding plugin)

0.058     2013-11-04 21:05:24 America/New_York

    [ADDED]

    - Added Prereqs::AuthorDeps

    [CHANGED]

    - Swap in RunExtraTests instead of CheckExtraTests

0.057     2013-10-20 14:15:20 America/New_York

    [INCOMPATIBLE CHANGES]

    - Requires v5 of Dist::Zilla and v4 of Pod::Weaver (currently
      only available as -TRIAL distributions)

    - Uses new SingleEncoding weaver plugin

    - Switches to ReadmeFromPod; currently needs 0.19_01 for dzil v5
      compatibility

0.056     2013-10-15 19:27:21 America/New_York

    [ADDED]

    - Added an "=requires" collector for a REQUIREMENTS sections (handy for
      roles or things with unusual external requirements)

    [CHANGED]

    - Made Pod weaving put USAGE before attribute/method/etc. collectors

    [REMOVED]

    - Removed '=usage' Pod::Weaver collector

    [DOCS]

    - Fixed typos and improved examples

0.055     2013-10-13 15:44:51 America/New_York

    [CHANGED]

    - Switches Test::Compile to use xt_mode (and sets minimum
      prereqs to support that)

    [PREREQS]

    - Bumps Test::ReportPrereqs version to 0.008 for warnings
      on unsatisfied prereqs

    - Bumps Pod::Wordlist version to 1.10 for many stoplist
      improvements
    
0.054     2013-09-23 16:19:18 America/New_York

    [CHANGED]

    - Dropped ExtUtils::MakeMaker configure_requires in generated
      distributions to 6.17 (the version in Perl v5.8.1)

0.053     2013-09-08 10:15:25 America/New_York

    [FIXED]

    - Let GithubMeta find 'user' from repository data
      (e.g. Perl-Toolchain-Gang)

    - Fix resource generation when not using github for issues

0.052     2013-09-08 09:47:50 America/New_York

    [PREREQS]

    - Bumped Dist::Zilla to 4.300038 for fixed MetaYML

lib/Pod/Weaver/PluginBundle/DAGOLDEN.pm
use strict;
use warnings;

package Pod::Weaver::PluginBundle::DAGOLDEN;

our $VERSION = '0.080';

use Pod::Weaver 4; # he played knick-knack on my door
use Pod::Weaver::Config::Assembler;

# Dependencies
use Pod::Weaver::Plugin::WikiDoc ();
use Pod::Elemental::Transformer::List 0.102000 ();
use Pod::Elemental::PerlMunger 0.200001        (); # replace with comment support
use Pod::Weaver::Section::Support 1.001        ();
use Pod::Weaver::Section::Contributors 0.008   ();

sub _exp { Pod::Weaver::Config::Assembler->expand_package( $_[0] ) }

my $repo_intro = <<'END';
This is open source software.  The code repository is available for
public review and contribution under the terms of the license.
END

my $bugtracker_content = <<'END';
Please report any bugs or feature requests through the issue tracker
at {WEB}.
You will be notified automatically of any progress on your issue.
END

sub mvp_bundle_config {
    my @plugins;
    push @plugins, (
        [ '@DAGOLDEN/SingleEncoding', _exp('-SingleEncoding'), {} ],
        [ '@DAGOLDEN/WikiDoc',        _exp('-WikiDoc'),        {} ],
        [ '@DAGOLDEN/CorePrep',       _exp('@CorePrep'),       {} ],
        [ '@DAGOLDEN/Name',           _exp('Name'),            {} ],
        [ '@DAGOLDEN/Version',        _exp('Version'),         {} ],

        [ '@DAGOLDEN/Prelude',     _exp('Region'),  { region_name => 'prelude' } ],
        [ '@DAGOLDEN/Synopsis',    _exp('Generic'), { header      => 'SYNOPSIS' } ],
        [ '@DAGOLDEN/Description', _exp('Generic'), { header      => 'DESCRIPTION' } ],
        [ '@DAGOLDEN/Usage',       _exp('Generic'), { header      => 'USAGE' } ],
        [ '@DAGOLDEN/Overview',    _exp('Generic'), { header      => 'OVERVIEW' } ],
        [ '@DAGOLDEN/Stability',   _exp('Generic'), { header      => 'STABILITY' } ],
    );

    for my $plugin (
        [ 'Requirements', _exp('Collect'), { command => 'requires' } ],
        [ 'Attributes',   _exp('Collect'), { command => 'attr' } ],
        [ 'Constructors', _exp('Collect'), { command => 'construct' } ],
        [ 'Methods',      _exp('Collect'), { command => 'method' } ],
        [ 'Functions',    _exp('Collect'), { command => 'func' } ],
      )
    {
        $plugin->[2]{header} = uc $plugin->[0];
        push @plugins, $plugin;
    }

    push @plugins,
      (
        [ '@DAGOLDEN/Leftovers', _exp('Leftovers'), {} ],
        [ '@DAGOLDEN/postlude', _exp('Region'), { region_name => 'postlude' } ],
        [
            '@DAGOLDEN/Support',
            _exp('Support'),
            {
                perldoc            => 0,
                websites           => 'none',
                bugs               => 'metadata',
                bugs_content       => $bugtracker_content,
                repository_link    => 'both',
                repository_content => $repo_intro
            }
        ],
        [ '@DAGOLDEN/Authors',      _exp('Authors'),      {} ],
        [ '@DAGOLDEN/Contributors', _exp('Contributors'), {} ],
        [ '@DAGOLDEN/Legal',        _exp('Legal'),        {} ],
        [ '@DAGOLDEN/List', _exp('-Transformer'), { 'transformer' => 'List' } ],
      );

    return @plugins;
}

# ABSTRACT: DAGOLDEN's default Pod::Weaver config
# COPYRIGHT

1;

=for Pod::Coverage mvp_bundle_config

=head1 DESCRIPTION

This is a L<Pod::Weaver> PluginBundle.  It is roughly equivalent to the
following weaver.ini:

  [-WikiDoc]

  [@Default]

  [Support]
  perldoc = 0
  websites = none
  bugs = metadata
  bugs_content = ... stuff (web only, email omitted) ...
  repository_link = both
  repository_content = ... stuff ...

  [Contributors]

  [-Transformer]
  transformer = List

=head1 USAGE

This PluginBundle is used automatically with the C<@DAGOLDEN> L<Dist::Zilla>
plugin bundle.

It also has region collectors for:

=for :list
* requires
* construct
* attr
* method
* func

=head1 SEE ALSO

=for :list
* L<Pod::Weaver>
* L<Pod::Weaver::Plugin::WikiDoc>
* L<Pod::Elemental::Transformer::List>
* L<Pod::Weaver::Section::Contributors>
* L<Pod::Weaver::Section::Support>
* L<Dist::Zilla::Plugin::PodWeaver>

=cut


Makefile.PL
use strict;
use warnings;

use 5.008001;

use ExtUtils::MakeMaker 6.17;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist::Zilla configuration the way DAGOLDEN does it",
  "AUTHOR" => "David Golden <dagolden\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.17"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-DAGOLDEN",
  "LICENSE" => "apache",
  "MIN_PERL_VERSION" => "5.008001",
  "NAME" => "Dist::Zilla::PluginBundle::DAGOLDEN",
  "PREREQ_PM" => {
    "Dist::Zilla" => "5.014",
    "Dist::Zilla::Plugin::Authority" => "1.006",
    "Dist::Zilla::Plugin::Bugtracker" => "1.110",
    "Dist::Zilla::Plugin::BumpVersionAfterRelease" => "0.003",
    "Dist::Zilla::Plugin::CPANFile" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckMetaResources" => "0.001",
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => "0.002",
    "Dist::Zilla::Plugin::CopyFilesFromBuild::Filtered" => 0,
    "Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch" => 0,
    "Dist::Zilla::Plugin::Git::Contributors" => "0.007",
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => "0.36",
    "Dist::Zilla::Plugin::InsertCopyright" => "0.001",
    "Dist::Zilla::Plugin::MetaNoIndex" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14",
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::Pod2Readme" => 0,
    "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
    "Dist::Zilla::Plugin::PromptIfStale" => "0.011",
    "Dist::Zilla::Plugin::RewriteVersion" => 0,
    "Dist::Zilla::Plugin::RunExtraTests" => 0,
    "Dist::Zilla::Plugin::SurgicalPodWeaver" => 0,
    "Dist::Zilla::Plugin::TaskWeaver" => "0.101620",
    "Dist::Zilla::Plugin::Test::Compile" => "2.036",
    "Dist::Zilla::Plugin::Test::MinimumVersion" => "2.000003",
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::PodSpelling" => "2.006001",
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => "0.016",
    "Dist::Zilla::Plugin::Test::Version" => 0,
    "Dist::Zilla::PluginBundle::Filter" => 0,
    "Dist::Zilla::PluginBundle::Git" => "1.121010",
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Moose" => "0.99",
    "Pod::Elemental::PerlMunger" => "0.200001",
    "Pod::Elemental::Transformer::List" => "0.102000",
    "Pod::Weaver" => 4,
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::Section::Contributors" => "0.008",
    "Pod::Weaver::Section::Support" => "1.001",
    "Pod::Wordlist" => "1.10",
    "Test::Portability::Files" => "0.06",
    "namespace::autoclean" => "0.09",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.080",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla" => "5.014",
  "Dist::Zilla::Plugin::Authority" => "1.006",
  "Dist::Zilla::Plugin::Bugtracker" => "1.110",
  "Dist::Zilla::Plugin::BumpVersionAfterRelease" => "0.003",
  "Dist::Zilla::Plugin::CPANFile" => 0,
  "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
  "Dist::Zilla::Plugin::CheckMetaResources" => "0.001",
  "Dist::Zilla::Plugin::CheckPrereqsIndexed" => "0.002",
  "Dist::Zilla::Plugin::CopyFilesFromBuild::Filtered" => 0,
  "Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch" => 0,
  "Dist::Zilla::Plugin::Git::Contributors" => "0.007",
  "Dist::Zilla::Plugin::Git::NextVersion" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => "0.36",
  "Dist::Zilla::Plugin::InsertCopyright" => "0.001",
  "Dist::Zilla::Plugin::MetaNoIndex" => 0,
  "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14",
  "Dist::Zilla::Plugin::MinimumPerl" => 0,
  "Dist::Zilla::Plugin::Pod2Readme" => 0,
  "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
  "Dist::Zilla::Plugin::PromptIfStale" => "0.011",
  "Dist::Zilla::Plugin::RewriteVersion" => 0,
  "Dist::Zilla::Plugin::RunExtraTests" => 0,
  "Dist::Zilla::Plugin::SurgicalPodWeaver" => 0,
  "Dist::Zilla::Plugin::TaskWeaver" => "0.101620",
  "Dist::Zilla::Plugin::Test::Compile" => "2.036",
  "Dist::Zilla::Plugin::Test::MinimumVersion" => "2.000003",
  "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
  "Dist::Zilla::Plugin::Test::PodSpelling" => "2.006001",
  "Dist::Zilla::Plugin::Test::Portability" => 0,
  "Dist::Zilla::Plugin::Test::ReportPrereqs" => "0.016",
  "Dist::Zilla::Plugin::Test::Version" => 0,
  "Dist::Zilla::PluginBundle::Filter" => 0,
  "Dist::Zilla::PluginBundle::Git" => "1.121010",
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "Moose" => "0.99",
  "Path::Tiny" => 0,
  "Pod::Elemental::PerlMunger" => "0.200001",
  "Pod::Elemental::Transformer::List" => "0.102000",
  "Pod::Weaver" => 4,
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::WikiDoc" => 0,
  "Pod::Weaver::Section::Contributors" => "0.008",
  "Pod::Weaver::Section::Support" => "1.001",
  "Pod::Wordlist" => "1.10",
  "Test::DZil" => 0,
  "Test::More" => "0.96",
  "Test::Portability::Files" => "0.06",
  "namespace::autoclean" => "0.09",
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
